ignite/handlers/early_stopping.py

```diff
@@ -55,7 +62,9 @@ def __call__(self, engine):
 
         if self.best_score is None:
             self.best_score = score
-        elif score <= self.best_score:
+        elif score - self.best_score <= self.min_delta:
+            if score > self.best_score:
```

Comment:
Should we store new score as best one in this case ?
For example, `min_delta=0.1`, 
- the last best score is `3.0` and current score is now `3.05`  => counter += 1 and best score is `3.05`
- the last best score is `3.05` and current score is now `3.1`  => counter += 1 and best score is `3.1`
- the last best score is `3.1` and current score is now `3.15`  => counter += 2 and best score is `3.15`
And we see that in the 3rd evaluation the model has improved to 3.15 from 3.0 with min_delta=0.1 but counter still incremented.
What do you think ?

Reply:
It depends whether `min_delta` should be a minimum increase in one step or a cumulative minimum increase since last patience reset. I assumed the former but it could be the latter as well. I've clarified a description in doc string.

Reply:
Maybe we could an additional flag to make both options possible ?

Reply:
I've added `cumulative_delta` flag and two more unit tests.